rmgpy/data/base.py

```diff
@@ -1267,17 +1267,27 @@ def isMoleculeForbidden(self, molecule):
         contains forbidden functionality, or ``False`` if not. Labeled atoms
         on the forbidden structures and the molecule are honored.
         """
+        from rmgpy.species import Species
+
         for entry in self.entries.values():
-            entryLabeledAtoms = entry.item.getLabeledAtoms()
-            moleculeLabeledAtoms = molecule.getLabeledAtoms()
-            initialMap = {}
-            for label in entryLabeledAtoms:
-                # all group labels must be present in the molecule
-                if label not in moleculeLabeledAtoms: break  
-                initialMap[moleculeLabeledAtoms[label]] = entryLabeledAtoms[label]
-            else:
-                if molecule.isMappingValid(entry.item, initialMap) and molecule.isSubgraphIsomorphic(entry.item, initialMap):
+            if isinstance(entry.item, Molecule) or isinstance(entry.item, Species):
```

Comment:
Perhaps we could first check equality between `len(molecule.vertices)` before entering the other conditions?
I understand we check fingerprints inside Molecule.isIsomorphic(), but it might still save us a bit, since most comparisons will have a different number of atoms than entry.item

Reply:
I think I'll leave it like this for now. I did some time trials, and it didn't seem like adding that check gave that much speedup. I do plan on revisiting this in the near future though, since I'm considering replacing this with an InChI comparison.